internal/servers/worker/status.go

```diff
@@ -99,6 +59,37 @@ func (w *Worker) LastStatusSuccess() *LastStatusInformation {
 	return w.lastStatusSuccess.Load().(*LastStatusInformation)
 }
 
+// WaitForNextSuccessfulStatusUpdate waits for the next successful status. It's
+// used by testing (and in the future, shutdown) in place of a more opaque and
+// possibly unnecessarily long sleep for things like initial controller
+// check-in, etc.
+//
+// The timeout is aligned with the worker's status grace period. A nil error
+// means the status was sent successfully.
+func (w *Worker) WaitForNextSuccessfulStatusUpdate() error {
+	w.logger.Debug("waiting for next status report to controller")
+	waitStatusStart := time.Now()
+	ctx, cancel := context.WithTimeout(w.baseContext, w.conf.StatusGracePeriodDuration)
+	defer cancel()
+	for {
+		select {
```

Comment:
Seems to be a good use-case for using a [Cond](https://golang.org/pkg/sync/#Cond), where the worker can just broadcast every time it performs an update to the `lastStatusSuccess` instead of polling every second.  Ok keeping it like this as well if that causes a larger refactor that seems.

Reply:
Let's leave this like it is right now.  I think adjusting the status check inter-communication in such a way would be a bit more impacting than just adjusting things here, in addition to me not necessarily understanding sync.Cond well (and I haven't seen many good examples of its use). Could be something we entertain in the future.